server/src/main/java/communication/Server.java

```diff
@@ -15,20 +17,20 @@
  */
 public class Server {
 
-  private static final Logger LOGGER = Logger.getLogger(Server.class.getName());
+  private ServerLogger serverLogger = ServerLogger.getInstance();
```

Comment:
Why do you have two interfaces: 
-ClientLogger 
-ServerLogger 
instead of MyLogger with 2 different implementations in shared module?